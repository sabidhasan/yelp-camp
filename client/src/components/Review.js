import React from 'react'
import PropTypes from 'prop-types'
import XButton from './XButton'
import RatingBar from './RatingBar'
import { formatDate } from '../helpers/helpers'

const Review = (props, context) => (
    <li className='Review'>
      <span className='Review__identity bold flex-center'>
        <img className='Review__image' src={props.photoURL} alt='User' />
        <span className='Review__author'>{props.displayName}</span>
      </span>
      <div className='Review__rating'>
        <RatingBar className='Review__rating-bar' rating={props.rating} small={true} />
        <span className='Review__time-stamp'>{formatDate(new Date(props.time))}</span>
      </div>
      {context.user && context.user.uid === props.uid ?
        <XButton className='Review__delete bold' onClick={() => props.deleteReview(props.id)} />
      : null}
      <span className='Review__text'>{props.text}</span>
    </li>
  )

Review.contextTypes = {
  user: PropTypes.object
};

export default Review;
