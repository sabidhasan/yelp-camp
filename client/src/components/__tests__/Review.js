import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Review from '../Review'

configure({ adapter: new Adapter() });

describe('<Review />', () => {
  let component;
  let props;
  let deleteReview;

  beforeEach(() => {
    deleteReview = jest.fn();
    props = {
      photoURL: 'URL',
      displayName: 'Some Name',
      rating: 3.2,
      time: 'Tuesday April 10, 2018',
      uid: '10',
      text: 'Text',
      deleteReview
    }
    component = shallow(<Review { ...props } />, {context: {user: {uid: 50}}})
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Review__delete now shown when uid different from props ID', () => {
    expect(component.find('.Review__delete')).toHaveLength(0)
  })

  it('The delete function is called when Review__delete is clicked', () => {
    component.setContext({
      user: {uid: '10'}
    });
    component.find('.Review__delete').simulate('click')
    expect(deleteReview).toHaveBeenCalledTimes(1)
  });

  it('Time is properly formattted', () => {
    expect(component.find('.Review__time-stamp').text()).toBe('10 April 2018')
  });

  it('Review component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
