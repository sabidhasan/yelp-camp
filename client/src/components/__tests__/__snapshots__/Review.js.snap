// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> Review component snapshot matches what is expected 1`] = `
<li
  className="Review"
>
  <span
    className="Review__identity bold flex-center"
  >
    <img
      alt="User"
      className="Review__image"
      src="URL"
    />
    <span
      className="Review__author"
    >
      Some Name
    </span>
  </span>
  <div
    className="Review__rating"
  >
    <RatingBar
      className="Review__rating-bar"
      rating={3.2}
      small={true}
    />
    <span
      className="Review__time-stamp"
    >
      10 April 2018
    </span>
  </div>
  <span
    className="Review__text"
  >
    Text
  </span>
</li>
`;
